dev-python/numpy/numpy-1.18.3.ebuild

```diff
@@ -0,0 +1,155 @@
+		https://numpy.org/doc/$(ver_cut 1-2 ${DOC_PV})/numpy-ref.pdf -> numpy-ref-${DOC_PV}.pdf
+		https://numpy.org/doc/$(ver_cut 1-2 ${DOC_PV})/numpy-user.pdf -> numpy-user-${DOC_PV}.pdf
+	)"
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~x86"
+IUSE="doc lapack"
+
+RDEPEND="
+	lapack? (
+		>=virtual/cblas-3.8
+		>=virtual/lapack-3.8
+	)
+"
+BDEPEND="
+	${RDEPEND}
+	app-arch/unzip
+	>=dev-python/cython-0.29.15[${PYTHON_USEDEP}]
+	lapack? (
+		>=sys-devel/automake-1.16.2
+		>=sys-devel/gcc-6.5.0-r1[fortran]
+		virtual/pkgconfig
+	)
+	test? (
+		>=dev-python/hypothesis-5.8.0[${PYTHON_USEDEP}]
+		>=dev-python/pytz-2019.3[${PYTHON_USEDEP}]
+		>=dev-python/cffi-1.14.0[${PYTHON_USEDEP}]
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-1.17.4-no-hardcode-blasv2.patch
+)
+
+distutils_enable_tests pytest
+
+src_unpack() {
+	default
+	if use doc; then
+		unzip -qo "${DISTDIR}"/numpy-html-${DOC_PV}.zip -d html || die
+	fi
+}
+
+src_prepare() {
+	default
+	eautomake
+}
+
+python_prepare_all() {
+	if use lapack; then
+		local incdir="${EPREFIX}"/usr/include
+		local libdir="${EPREFIX}"/usr/$(get_libdir)
+		cat >> site.cfg <<-EOF || die
+			[blas]
+			include_dirs = ${incdir}
+			library_dirs = ${libdir}
+			blas_libs = cblas,blas
+			[lapack]
+			library_dirs = ${libdir}
+			lapack_libs = lapack
+		EOF
+	else
+		export {ATLAS,PTATLAS,BLAS,LAPACK,MKL}=None
+	fi
+
+	export CC="$(tc-getCC) ${CFLAGS}"
+
+	append-flags -fno-strict-aliasing
+
+	# See progress in http://projects.scipy.org/scipy/numpy/ticket/573
+	# with the subtle difference that we don't want to break Darwin where
+	# -shared is not a valid linker argument
+	if [[ ${CHOST} != *-darwin* ]]; then
+		append-ldflags -shared
+	fi
+
+	# only one fortran to link with:
+	# linking with cblas and lapack library will force
+	# autodetecting and linking to all available fortran compilers
+	append-fflags -fPIC
+	if use lapack; then
+		NUMPY_FCONFIG="config_fc --noopt --noarch"
+		# workaround bug 335908
+		[[ $(tc-getFC) == *gfortran* ]] && NUMPY_FCONFIG+=" --fcompiler=gnu95"
+	fi
+
+	# don't version f2py, we will handle it.
+	sed -i -e '/f2py_exe/s: + os\.path.*$::' numpy/f2py/setup.py || die
+
+	# disable fuzzed tests
+	find numpy/*/tests -name '*.py' -exec sed -i \
+		-e 's:def \(.*_fuzz\):def _\1:' {} + || die
+	# very memory- and disk-hungry
+	sed -i -e 's:test_large_zip:_&:' numpy/lib/tests/test_io.py || die
+
+	distutils-r1_python_prepare_all
+}
+
+python_compile() {
+	export MAKEOPTS=-j1 #660754
+
+	local python_makeopts_jobs=""
+	distutils-r1_python_compile \
+		${python_makeopts_jobs} \
+		${NUMPY_FCONFIG}
+}
+
+python_test() {
+	distutils_install_for_testing --single-version-externally-managed \
+		--record "${TMPDIR}/record.txt" ${NUMPY_FCONFIG}
+
+	cd "${TMPDIR}" || die
+
+	"${EPYTHON}" -c "
+import numpy, sys
+r = numpy.test(label='full', verbose=3)
+sys.exit(0 if r else 1)" || die "Tests fail with ${EPYTHON}"
+}
+
+python_install() {
+	distutils-r1_python_install ${NUMPY_FCONFIG}
```

Comment:
this is missing the fix for missing files.

Reply:
Added the fix from https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=1b53d722d793c11f16e81bc2c468cbb74d0c8a32 .